h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -2780,7 +2806,23 @@ public void setCacheSize(int mb) {
     }
 
     public boolean isClosed() {
-        return closed;
+        if (state == STATE_OPEN) {
+            return false;
+        }
+        int millis = 1;
```

Comment:
probably best to comment why we're doing this so future people do not attempt to "optimize" it